wlct/management/commands/engine.py

```diff
@@ -72,6 +74,94 @@ def parse_and_update_clan_logo():
     except Exception:
         log_exception()
 
+def is_correct_player(player_token, player_team):
+    try:
+        player = Player.objects.filter(token=player_token)[0]
+        tt = TournamentTeam.objects.filter(pk=player_team)[0]
+        if player.clan and player.clan.name == tt.clan_league_clan.clan.name:
+            return True
+
+        tplayers = TournamentPlayer.objects.filter(team=tt)
+        for tplayer in tplayers:
+            if tplayer.player.token == player_token:
+                return True
+        return False
+    except:
+        log_exception()
+        return False
+
+
+def is_valid_team_order(team_order, team_dict, teams):
+    for i in range(len(team_order)):
+        for player in team_dict[team_order[i]]:
+            if is_correct_player(player, teams[i]):
+                return True
+    return False
+
+def validate_cl_tournament_games():
+    try:
+        print("Running validate_cl_tournament_games on thread {}".format(threading.get_ident()))
+        tournament_games = TournamentGame.objects.filter(is_finished=False)
```

Comment:
Need to pass in the round robin tournament the game is in to filter on as well, otherwise this looks at ALL unfinished games.

Reply:
You can write a quick helper for this and then turn this into a loop.

def get_clan_league_tournaments():
    cl = ClanLeague.objects.filter(clan_league_template__isnull=False, clan_league_division__isnull=False, is_official=True)

    tournaments = ClanLeagueTournament.objects.filter(parent_tournament=cl[0])
    
    loop through tournaments processing all games in them...

Reply:
Added that in now. Instead of the filters on the CL object, I opted to use the has_started and is_official flag